ui/nuxeo-data-table/data-table-cell.js

```diff
@@ -75,6 +76,23 @@ import './data-table-templatizer-behavior.js';
     ready() {
       super.ready();
       this.setAttribute('role', 'cell');
+
+      if (this.header) {
+        // distribute the space evenly between all nodes in the header cell
+        this._observer = new FlattenedNodesObserver(this, ({ addedNodes }) => {
+          const sort = addedNodes.find((node) => node instanceof Nuxeo.DataTableColumnSort);
+          if (sort) {
+            const nodes = addedNodes.filter((node) => node !== sort && node.style);
+            if (nodes.length > 0) {
+              const cellMaxWidth = 100 / nodes.length;
+              const sortOffset = sort.getComputedStyleValue('width').match(/[\f\d]+/) / nodes.length;
+              nodes.forEach((node) => {
+                node.style.width = `calc(${cellMaxWidth}% - ${sortOffset}px)`;
+              });
+            }
+          }
+        });
+      }
```

Comment:
I feel that we are just fighting CSS here with javascript, because of a quirk where flex items (in this case `nuxeo-dropdown-aggregation` and `nuxeo-data-table-column-sort`) won't shrink below content size, but I think we can work around this with a `min-content: 0`.

I tested it locally but not thoroughly, could you take it for a spin?
```
          ::slotted(*) {
            min-width: 0;
          }

          ::slotted(nuxeo-data-table-column-sort) {
            flex-basis: 0;
          }
```
or something more conservative, like:
```
          ::slotted(nuxeo-dropdown-aggregation) {
            min-width: 0;
          }

          ::slotted(nuxeo-data-table-column-sort) {
            flex-basis: 0;
          }
```
@semisse do you know of any other workarounds for this kind of stuff? Not sure if this covers everthing.

Reply:
@Gabez0r I tested this locally and seems to work as expected. This is probably better than my approach of using javascript to fight CSS 😬  If we agree in this solution, I'll drop the first two commits and we keep the last one. Let me know what you think (and @semisse too).

Reply:
If it works, I'd prefer to go with the css solution (and drop the other two commits) 🤞 .

Reply:
@Gabez0r It seems that we don't need the `flex-basis`, and this breaks the functional tests https://github.com/nuxeo/nuxeo-web-ui/actions/runs/1841428842

```
::slotted(nuxeo-data-table-column-sort) {
  flex-basis: 0;
}
```


Any reason why you included the `flex-basis` in the first place?

Reply:
![ELEMENTS-1458](https://user-images.githubusercontent.com/738700/153893466-2628bb4f-f430-46f9-ae47-215e797ac02a.gif)

Reply:
To set the initial size of the item to 0, instead of the default oversize initial value it has by default. But feel free to play with it. In my experiments, removing it didn't solve the issue but I didn't spend to much time with it.

Reply:
In my tests, just keeping the `min-width: 0` works and solves the issue, but I'm spending some more time to test it (and run the ftests as well).